tests/test_spark.py

```diff
@@ -156,3 +158,21 @@ def test_get_pings_none_filter(test_store, dummy_pool_executor,
     pings = get_pings(spark_context, app='*')
 
     assert sorted(pings.collect()) == ['value1', 'value2']
+
+
+@pytest.mark.skip("monkeypatching doesn't persist across spark context")
+def test_get_heka_pings(data_dir, test_store, dummy_pool_executor, monkeypatch,
+                        spark_context):
+    from moztelemetry.heka.message_parser import _parse_json
+    mock_parse_json = MagicMock(name='_parse_json',
+                                wraps=_parse_json)
+    import moztelemetry.heka.message_parser
+    monkeypatch.setattr(moztelemetry.heka.message_parser, '_parse_json', mock_parse_json)
+
+    heka_content = open("{}/test_telemetry_gzip.heka".format(data_dir), "rb").read()
```

Comment:
If you want to run an end-to-end test like this you should probably use https://github.com/spulec/moto#stand-alone-server-mode. To make it work with Dataset, you need to modify S3Store so that you can pass a custom `endpoint_url` to `boto3.resource` and `boto3.client`.